src/Plutarch/Test/QuickCheck.hs

```diff
@@ -213,60 +326,120 @@ type family PLamArgs (p :: S -> Type) :: [Type] where
 class
     (PLamArgs p ~ args) =>
     HaskEquiv
+        (e :: Equality)
+        (par :: Partiality)
         (h :: Type)
         (p :: S -> Type)
         (args :: [Type])
     where
     haskEquiv :: h -> TestableTerm p -> NP Gen args -> Property
```

Comment:
Given that we're moving away from SOP generics in general, is this going to pose a problem?

Reply:
Considering that we are eventually going to put this into `Plutarch` repo itself, this should be okay. I think. After all, `HaskEquiv` exists in current `plutarch-test` for hedgehog, and it uses SOP as well.

Reply:
Fair enough.